apollo-federation/src/operation/mod.rs

```diff
@@ -3292,20 +3342,39 @@ impl FieldSelection {
         }))
     }
 
+    fn with_updated_element(&self, element: FieldData) -> Self {
+        Self {
+            field: Field::new(element),
+            ..self.clone()
+        }
+    }
+
     fn normalize(
         &self,
         parent_type: &CompositeTypeDefinitionPosition,
         named_fragments: &NamedFragments,
         schema: &ValidFederationSchema,
         option: NormalizeSelectionOption,
     ) -> Result<Option<SelectionOrSet>, FederationError> {
+        let field_position = if self.field.parent_type_position() == *parent_type {
```

Comment:
in JS side of things those elements hold references to the schema but on RS those are just type names so we need to explicitly check for the schema equality as well